### PR TITLE
GuniProxyActions and Exchange contracts update

### PR DESCRIPTION
**Pull request by fenris85**
# [Use newly deployed exchange contract in GUNI open](https://app.shortcut.com/oazo-apps/story/3159/use-newly-deployed-exchange-contract-in-guni-open)

## Changes 👷‍♀️
- Updated GuniProxyActions contract to the latest version with events emit.
- Added third Exchange contract with 0.04% fee tier. 
- Refactor pipelines to use different fees

## How to test 🧪
  <Please explain how to test your changes>
- step 1 ...
    
## Definition of done ✔️

- [ ] Acceptance criteria for each issue met
- [ ] Unit tests written where needed and passing
- [ ] End-to-end tests for happy path
- [ ] Documentation updated <When applicable>
- [ ] Project builds without errors
- [ ] Non-functional requirements met
- [ ] Code reviewed and functionality tested by the reviewer (locally, Heroku, etc.)
- [ ] Feature verified and accepted by Product Owner
- [ ] Project deployed to production environment

### CODE REVIEW

File: blockchain/calls/proxyActions.ts [hunk 1]
```diff
@@ -491,16 +485,9 @@ export type MultiplyAdjustData = {
...
-    mcdJug,
-    dssCdpManager,
-    dssMultiplyProxyActions,
-    tokens,
-    exchange,
-    fmm,
-  } = context
+  const { contract, joins, mcdJug, dssCdpManager, dssMultiplyProxyActions, tokens, fmm } = context
+
+  const exchange = context['defaultExchange']
```
Comment (paszkowskiDamian) [inline blockchain/calls/proxyActions.ts:490]: Can't you just take `defaultExchange` from context like `const { defaultExchange } = context` with the rest

File: features/exchange/exchange.ts [hunk 2]
```diff
@@ -170,35 +170,17 @@ export function createExchangeQuote$(
   slippage: BigNumber,
   amount: BigNumber,
   action: ExchangeAction,
+  exchangeType: string,
```
Comment (paszkowskiDamian) [inline features/exchange/exchange.ts:173]: Can you add type `'defaultExchange' | 'noFeesExchange' | 'lowerFeesExchange'` so it's easer to find what kind of exchanges we have

**Comment by paszkowskiDamian** [comment]
For some reason for this gUni token net value is negative. I think this is due to ilk not being released yet. 

image

**Comment by fenris85** [comment]
> For some reason for this gUni token net value is negative. I think this is due to ilk not being released yet.
> 
> image

yeah, I think it's because ilk params are not set properly yet by the spell. The same is for stability fee:
image

**Comment by paszkowskiDamian** [comment]
Exactly that I'll not worry about it :) 

But I guess this hack in calculation needs to be adjusted so the pipeline uses actual liquidation ratio not hardcoded.

**Review by paszkowskiDamian — APPROVED** [review]

**Review by paszkowskiDamian — APPROVED** [review]

**Pull request merged**